internal/connectors/plugins/public/modulr/accounts.go

```diff
@@ -27,56 +29,33 @@ func (p Plugin) fetchNextAccounts(ctx context.Context, req models.FetchNextAccou
 	}
 
 	var accounts []models.PSPAccount
+	needMore := false
 	hasMore := false
 	for page := 0; ; page++ {
 		pagedAccounts, err := p.client.GetAccounts(ctx, page, req.PageSize, oldState.LastCreatedAt)
 		if err != nil {
 			return models.FetchNextAccountsResponse{}, err
 		}
 
-		if len(pagedAccounts.Content) == 0 {
-			break
-		}
-
-		for _, account := range pagedAccounts.Content {
-			createdTime, err := time.Parse("2006-01-02T15:04:05.999-0700", account.CreatedDate)
-			if err != nil {
-				return models.FetchNextAccountsResponse{}, err
-			}
-
-			switch createdTime.Compare(oldState.LastCreatedAt) {
-			case -1, 0:
-				// Account already ingested, skip
-				continue
-			default:
-			}
-
-			raw, err := json.Marshal(account)
-			if err != nil {
-				return models.FetchNextAccountsResponse{}, err
-			}
-
-			accounts = append(accounts, models.PSPAccount{
-				Reference:    account.ID,
-				CreatedAt:    createdTime,
-				Name:         &account.Name,
-				DefaultAsset: pointer.For(currency.FormatAsset(supportedCurrenciesWithDecimal, account.Currency)),
-				Raw:          raw,
-			})
-
-			newState.LastCreatedAt = createdTime
-
-			if len(accounts) >= req.PageSize {
-				break
-			}
+		accounts, err = fillAccounts(pagedAccounts, accounts, oldState)
+		if err != nil {
+			return models.FetchNextAccountsResponse{}, err
 		}
 
-		if len(accounts) >= req.PageSize {
-			hasMore = true
+		needMore, hasMore = pagination.ShouldFetchMore(accounts, pagedAccounts, req.PageSize)
+		if !needMore || !hasMore {
```

Comment:
I think we still want to set the cache even if `hasMore` is false, or else we'll try to fetch the final batch again.

`hasMore` is a variable that affects whether temporal retries or not, but shouldn't need to affect the flow of this loop.

Reply:
Done !